File: Hayaa.Security.Service/Client/C#/Hayaa.WorkerSecurity.Client/WorkerSecurityProvider.cs
using Hayaa.CacheKeyStatic;
using Hayaa.Redis.Client;
using Hayaa.Security.Service;
using System;

namespace Hayaa.WorkerSecurity.Client
{
    internal class WorkerSecurityProvider
    {
        public static Boolean UserAuth(String token)
        {
            Boolean result = false;
            var info = RedisService.Get<LoginToken>(DefineTable.CacheName, String.Format(UserAuthorityCacheKey.AuthorityCacheKey, token));
            if (info != null)
            {
                if ((info.Status) && (info.EndTime > DateTime.Now))
                {
                    result = true;
                }

            }
            return result;
        }
        public static Boolean WorkerAuth(String token)
        {
            Boolean result = false;
            var info = RedisService.Get<JobToken>(DefineTable.CacheName, String.Format(JobAuthorityCacheKey.AuthorityCacheKey, token));
            if (info != null)
            {
                if(info.EndTime > DateTime.Now)
                {
                    result = true;
                }

            }
            return result;
        }
    }
}

File: v1/Client/Java/hayaa.security.client/hayaa.worker.security.client/src/main/java/hayaa/workersecurity/client/WorkerSecurityProvider.java
package hayaa.workersecurity.client;

import hayaa.cachekeystatic.JobAuthorityCacheKey;
import hayaa.cachekeystatic.UserAuthorityCacheKey;
import hayaa.redis.client.RedisService;

import java.util.Date;

class WorkerSecurityProvider {
    public static Boolean UserAuth(String token)
    {
        Boolean result = false;
        LoginToken info = RedisService.get(DefineTable.cacheName, UserAuthorityCacheKey.AuthorityCacheKey.replace("{0}",token));
        if (info != null)
        {
            if ((info.getStatus()) && (info.getEndTime().getTime() > (new Date()).getTime()))
            {
                result = true;
            }

        }
        return result;
    }
    public static Boolean WorkerAuth(String token)
    {
        Boolean result = false;
        JobToken info = RedisService.get(DefineTable.cacheName,JobAuthorityCacheKey.AuthorityCacheKey.replace("{0}",token));
        if (info != null)
        {
            if(info.getEndTime().getTime() > (new Date()).getTime())
            {
                result = true;
            }

        }
        return result;
    }
}
